File: engine/src/main/java/de/hanno/hpengine/shader/VertexShader.java
package de.hanno.hpengine.shader;

import de.hanno.hpengine.util.ressources.CodeSource;

import java.io.IOException;

public class VertexShader extends AbstractShader {

    public static VertexShader load(CodeSource sourceCode) throws IOException {
        return Shader.loadShader(VertexShader.class, sourceCode);
    }

    @Override
    public OpenGLShader getShaderType() {
        return OpenGLShader.VertexShader;
    }
}

File: engine/src/main/java/de/hanno/hpengine/engine/graphics/shader/VertexShader.kt
package de.hanno.hpengine.engine.graphics.shader

import de.hanno.hpengine.engine.graphics.shader.define.Defines
import de.hanno.hpengine.util.ressources.CodeSource
import de.hanno.hpengine.util.ressources.FileBasedCodeSource

class VertexShader private constructor(override var shaderSource: CodeSource, override var id: Int) : Shader {
    override val shaderType: Shader.ShaderType = Shader.ShaderType.VertexShader

    companion object {
        fun load(programManager: OpenGlProgramManager, sourceCode: FileBasedCodeSource, defines: Defines = Defines()): VertexShader {
            return VertexShader(sourceCode, programManager.loadShader(Shader.ShaderType.VertexShader, sourceCode, defines))
        }
    }
}